android/framework/cmake-config/PlatformConfig.cmake

```diff
@@ -51,9 +51,8 @@ configure_file("${GFXRECON_SOURCE_DIR}/project_version.h.in" "${CMAKE_BINARY_DIR
 add_library(platform_specific INTERFACE)
 target_compile_definitions(platform_specific INTERFACE _FILE_OFFSET_BITS=64 PAGE_GUARD_ENABLE_UCONTEXT_WRITE_DETECTION VK_USE_PLATFORM_ANDROID_KHR)
 
-add_library(vulkan_registry INTERFACE)
-target_include_directories(vulkan_registry INTERFACE ${GFXRECON_SOURCE_DIR}/external/Vulkan-Headers/include)
-target_compile_definitions(vulkan_registry INTERFACE VK_NO_PROTOTYPES VK_ENABLE_BETA_EXTENSIONS)
+add_subdirectory(${GFXRECON_SOURCE_DIR}/external/Vulkan-Headers)
```

Comment:
The `android/layer/CMakeLists.txt` includes several instances of add_subdirectory with explicit binary folders specified. I tried to model that in an effort to make things work. Not sure about it because the cmake code is run through `include(cmake-config/PlatformConfig.cmake)`, and I'm unsure how relative paths work in that context.

Reply:
```
Execution failed for task ':layer:generateJsonModelRelease'.
> /home/runner/work/gfxreconstruct/gfxreconstruct/android/layer/CMakeLists.txt : C/C++ release|armeabi-v7a : CMake Error at /home/runner/work/gfxreconstruct/gfxreconstruct/external/Vulkan-Headers/CMakeLists.txt:8 (cmake_minimum_required):
    CMake 3.15...3.25 or higher is required.  You are running version 3.10.2
    ```
Did the actions runner change??? Seems I have an invalid cmake version now for who knows what reason.

Reply:
OH I am being silly.
The Vulkan-Headers/CMakelists.txt requires CMake 3.15, but the android code is running using 3.10.2. Which means that either the CMake version in android is updated or the android code reverts to the old way (manually creating a library). The latter may be difficult if there is shared CMake code between android and the rest of the project.

Reply:
The `linux` build in `.github/workflows/ci_build.xml` invokes `    - uses: lukka/get-cmake@latest` under `steps:`.  I wonder if we need that for the android build as well?  @charles-lunarg you could try adding it as part of your PR, see what GH does.

Reply:
I believe that would work, but that would have to happen in conjunction with raising the minimum (that way we don't advertise a minimum lower than what we actually require). 

I would like to raise the minimum to 3.16.3, which is what top level CMakeLists.txt require.

Reply:
Seems that the ` - uses: lukka/get-cmake@latest` is not able to change the CMake version used. According to the android docs I found, 3.10.2 is the newest version provided in the NDK. They do mention being able to use newer CMake versions, you just have to provide the binary yourself.

Reply:
Does that mean we need to specify the `cmake  `binary to use in an environment variable or some such?  @juan-lunarg any thoughts or should we just wait until we're finished transitioning our Android build process?

Reply:
I have reverted the android cmake changes - seems that the CMake version provided by the NDK is too old for the Vulkan-headers cmake code. And because it doesn't make use of the version information, it is fine to keep as is.